pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,832 @@
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, iv_curve["i_gold"].tolist())) + "\n"
+
+    return gold_dataset_csv
+
+
+def save_gold_dataset(gold_dataset,
+                      json_rel_path="sdm.json",
+                      csv_rel_path="sdm.csv"):
+    """
+    Save gold dataset to file in two possible formats.
+
+    The gold dataset dict is saved in a lossless, machine-interoperable json
+    file and in a lossy, but human-readable, csv file.
+
+    Relative file paths are used, which are relative to the current directory.
+    Defaults are customizable. Set a file path to None to skip saving that
+    file.
+
+    Parameters
+    ----------
+    gold_dataset : dict
+
+    json_rel_path : str
+        Pass ``None`` to skip writing this file.
+
+    csv_rel_path :str
+        Pass ``None`` to skip writing this file.
+
+    Returns
+    -------
+
+    """
+
+    # When requested, write dataset in json file, skips when empty path string
+    if json_rel_path:
+        # This should normalize the path for the particular OS
+        with open(os.path.abspath(json_rel_path), 'w') as fp:
+            json.dump(gold_dataset, fp)
+
+    # When requested, write dataset in csv file, skips when empty path string
+    if csv_rel_path:
+        # This should normalize the path for the particular OS
+        with open(os.path.abspath(csv_rel_path), 'w') as fp:
+            fp.write(pretty_print_gold_dataset(gold_dataset))
+
+
+def load_gold_dataset(json_rel_path="sdm.json"):
+    """
+    Load gold dataset from disk.
+
+    The gold dataset is loaded from a json file into a dict in lossless,
+    machine-interoperable format. Pass this to convert_gold_dataset() in order
+    to use the data with pvlib on a specific machine.
+
+    Parameters
+    ----------
+    json_rel_path : str
+
+    Returns
+    -------
+    gold_dataset : dict
+    """
+
+    # This should normalize the path for the particular OS
+    with open(os.path.abspath(json_rel_path), 'r') as fp:
+        return json.load(fp)
+
+
+def gauge_gold_dataset(gold_dataset, test_func):
+    """
+    Gauge modeling functions for accuracy and speed against the gold dataset.
+
+    For each modeling function passed in the test_func dictionary, this gauges
+    and returns the I-V curve for each device with the least accuracy as
+    compared to the gold dataset. Gauges are for both the residual of the
+    current sum at the diode node and the residual in terminal I or V. Timing
+    data and summary statistics are also returned for the ensemble of
+    computations per function and per device.
+
+    This also helps verify that a given function meets the pvlib interface.
+
+    Parameters
+    ----------
+    gold_dataset : dict
+
+    test_func : dict
+
+    Returns
+    -------
+    results : list
+        A per-device list of results with each function tested/benchmarked. The
+worst computed I-V curve is returned along with performance timing information
```

Comment:
These lines should be indented as well, at least for consistency with the rest of the library. I am surprised ``flake8`` does not complain on my checkout of the pr.